sway-core/src/asm_generation/from_ir.rs

```diff
@@ -1602,39 +1603,50 @@ impl<'ir> AsmBuilder<'ir> {
         assert!(offset == 0);
         assert!(ptr_ty.eq(self.context, &Type::B256));
 
-        // Expect ptr_ty here to also be b256 and offset to be whatever...
-        let val_ptr = self.resolve_ptr(val);
-        if val_ptr.value.is_none() {
-            return val_ptr.map(|_| ());
-        }
-        let (val_ptr, ptr_ty, offset) = val_ptr.value.unwrap();
-
-        // Expect the ptr_ty for val to also be B256
-        assert!(ptr_ty.eq(self.context, &Type::B256));
+        let val_reg = if matches!(
+            &self.context.values[val.0].value,
+            ValueDatum::Instruction(Instruction::IntToPtr(..))
```

Comment:
Rather than special case `IntToPtr` here would it work instead to update `resolve_ptr()` to handle both `GetPointer` and `IntToPtr`?

Reply:
> Rather than special case `IntToPtr` here would it work instead to update `resolve_ptr()` to handle both `GetPointer` and `IntToPtr`?

I considered that, but there's some computation that's done later which isn't needed for `IntToPtr` (but needed if resolved as a stack pointer in `resolve_ptr`. So it wouldn't serve any purpose since I'll need to handle them differently here anyway.